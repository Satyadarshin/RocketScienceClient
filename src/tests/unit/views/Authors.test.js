/*
 * Authors view unit test with Jest.
 */
import faker from 'faker';
import Vue from 'vue';
import { shallowMount } from '@vue/test-utils'
import AuthorsView from '@/views/Authors.vue';

/**
 * As we only wish to test our Authors Component, we can 'mock' our dependencies.
 * This essentially means creating a fake version of the dependency, which we are able to specify the behaviour of.
 */
const mockAuthors = [faker.random.word()];
jest.mock('@/services/sf-api-service.js', () => ({ fetchGenreCollection: jest.fn(x => Promise.resolve(mockAuthors)) }));

describe('Authors view', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(AuthorsView, {
      // Props to pass in for this test
      propsData: {
        spotlightOn: '',
      },
      // Stubs allows us to test this component without having to worry about its child components.
      stubs: ['router-view', 'app-author-table'],
    })
  });

  it('Snapshot', ()=> {
    expect(wrapper).toMatchSnapshot()
  })

  it('Should display the main heading', () => {
    // For every unit test there will be an expected result,
    // based on how we expect the code to behave if it is working correctly.
    const expected = 'Authors';
    // An actual result which we get from running the code.
    const actual = wrapper.find('h2').text();
    // We then compare the two to check if the code behaves as expected.
    expect(actual).toEqual(expected);
  });

  it('Should fetch the authors collection', async () => {
    // In some async scenarios, `nextTick` is necessary in order for view to carry out its internal updates.
    await Vue.nextTick();
    expect(wrapper.vm.authors).toEqual(mockAuthors);
  });
});
