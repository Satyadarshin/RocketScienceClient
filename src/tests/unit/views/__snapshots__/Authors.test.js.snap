// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `undefined`;

exports[`Authors view Snapshot 1`] = `
<div id="authors">
  <h2>Authors</h2>
  <p>This is a reference for SF authors of <em>significance</em>, that is an author that has either won or been nominated for a literary award for a text that is recgnised as Science Fiction.</p>
  <div class="md-layout">
    <div class="md-layout-item">
      <router-view-stub></router-view-stub>
    </div>
    <div class="md-layout-item">
      <app-author-table-stub authorlist="strategic"></app-author-table-stub>
    </div>
  </div>
</div>
`;
